sdk/storage/Azure.Storage.Blobs.AspNetCore.DataProtection/tests/AzureBlobXmlRepositoryTests.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Azure;
+using Azure.Core;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.AspNetCore.DataProtection;
+using Azure.Storage.Blobs.Models;
+using Moq;
+using NUnit.Framework;
+
+namespace Microsoft.AspNetCore.DataProtection.Azure.Storage.Blob
+{
+    public class AzureBlobXmlRepositoryTests
+    {
+        [Test]
+        public void StoreCreatesBlobWhenNotExist()
+        {
+            BlobRequestConditions uploadConditions = null;
+            byte[] bytes = null;
+            string contentType = null;
+
+            var mock = new Mock<BlobClient>();
+
+            mock.Setup(c => c.UploadAsync(
+                    It.IsAny<Stream>(),
+                    It.IsAny<BlobHttpHeaders>(),
+                    It.IsAny<IDictionary<string, string>>(),
+                    It.IsAny<BlobRequestConditions>(),
+                    It.IsAny<IProgress<long>>(),
+                    It.IsAny<AccessTier?>(),
+                    It.IsAny<StorageTransferOptions>(),
+                    It.IsAny<CancellationToken>()))
+                .Returns(async (Stream strm, BlobHttpHeaders headers, IDictionary<string, string> metaData, BlobRequestConditions conditions, IProgress<long> progress, AccessTier? access, StorageTransferOptions transfer, CancellationToken token) =>
+                {
+                    using var memoryStream = new MemoryStream();
+                    strm.CopyTo(memoryStream);
+                    bytes = memoryStream.ToArray();
+                    uploadConditions = conditions;
+                    contentType = headers?.ContentType;
+
+                    await Task.Yield();
+
+                    var mockResponse = new Mock<Response<BlobContentInfo>>();
+                    var blobContentInfo = BlobsModelFactory.BlobContentInfo(ETag.All, DateTimeOffset.Now.AddDays(-1), Array.Empty<byte>(), "", 1);
+
+                    mockResponse.Setup(c => c.Value).Returns(blobContentInfo);
+                    return mockResponse.Object;
+                });
+
+            var repository = new AzureBlobXmlRepository(mock.Object);
+            repository.StoreElement(new XElement("Element"), null);
+
+            Assert.AreEqual("*", uploadConditions.IfNoneMatch.ToString());
+            Assert.AreEqual("application/xml; charset=utf-8", contentType);
+            var element = "<Element />";
+
+            Assert.AreEqual(bytes, GetEnvelopedContent(element));
+        }
+
+        [Test]
+        public void StoreUpdatesWhenExistsAndNewerExists()
```

Comment:
Can we add one to test what happens when a blob does exist there but it's not ASP.NET keys?